### DISCUSSION TITLE
Missing dependencies on Skiasharp lib

### DISCUSSION THREAD

**Discussion by jstefcek** [QuestPDF/QuestPDF#431] [Q&A]
Hello, 
I am incredibly happy with your plugin, which helps to make me a genuinely nice PDF. When I run my program in Visual Studio, then there isn't any problem, but when I install the program on my notebook where the program should run with fails on: 

[QuestPDF.Drawing.Exceptions.InitializationException](http://questpdf.drawing.exceptions.initializationexception/): Cannot create the PDF document using the SkiaSharp-related library. This exception usually means that, on your operating system where you run the application, SkiaSharp-related requires installing additional dependencies. Such dependencies are available as additional nuget packages, for example *.NativeAssets.Linux.NoDependencies. Some operating systems may require installing multiple nugets, e.g. MacOS may need both *.NativeAssets.macOS.NoDependencies and *.NativeAssets.Linux.NoDependencies.Please refer to the SkiaSharp-related documentation for more details. Also, please consult the inner exception that has been originally thrown by the dependency library. ---> System.TypeInitializationException: Inicializační metoda typu SkiaSharp.SKAbstractManagedWStream vyvolala výjimku. ---> System.DllNotFoundException: Unable to load library 'libSkiaSharp'. v [SkiaSharp.LibraryLoader.LoadLocalLibrary[T](String](http://skiasharp.libraryloader.loadlocallibrary[t]%28string/) libraryName) v D:\a\1\s\binding\Binding.Shared\LibraryLoader.cs:řádek 38 v SkiaSharp.SkiaApi.<>c.<.cctor>b__1831_0() v D:\a\1\s\binding\Binding\SkiaApi.cs:řádek 15 v System.Lazy`1.CreateValue() v System.Lazy`1.LazyInitValue() v System.Lazy`1.get_Value() v [SkiaSharp.SkiaApi.sk_managedwstream_set_procs(SKManagedWStreamDelegates](http://skiasharp.skiaapi.sk_managedwstream_set_procs%28skmanagedwstreamdelegates/) procs) v D:\a\1\s\binding\Binding\SkiaApi.generated.cs:řádek 13300 v SkiaSharp.SKAbstractManagedWStream..cctor() v D:\a\1\s\binding\Binding\SKAbstractManagedWStream.cs:řádek 22 --- Konec trasování zásobníku pro vnitřní výjimku --- v SkiaSharp.SKAbstractManagedWStream..ctor(Boolean owns) v [SkiaSharp.SKDocument.CreatePdf(Stream](http://skiasharp.skdocument.createpdf%28stream/) stream, SKDocumentPdfMetadata metadata) v D:\a\1\s\binding\Binding\SKDocument.cs:řádek 138 v [QuestPDF.Drawing.PdfCanvas.CreatePdf(Stream](http://questpdf.drawing.pdfcanvas.createpdf%28stream/) stream, DocumentMetadata documentMetadata) --- Konec trasování zásobníku pro vnitřní výjimku --- v [QuestPDF.Drawing.PdfCanvas.CreatePdf(Stream](http://questpdf.drawing.pdfcanvas.createpdf%28stream/) stream, DocumentMetadata documentMetadata) v [QuestPDF.Drawing.DocumentGenerator.GeneratePdf(Stream](http://questpdf.drawing.documentgenerator.generatepdf%28stream/) stream, IDocument document) v [QuestPDF.Fluent.GenerateExtensions.GeneratePdf(IDocument](http://questpdf.fluent.generateextensions.generatepdf%28idocument/) document, String filePath) v [DecinkaApp.createPDF.Template(String](http://decinkaapp.createpdf.template%28string/) test) v [DecinkaApp.Dialog_Screens.Add_Vycetka_Gen_PDF.btn_odeslat_Click(Object](http://decinkaapp.dialog_screens.add_vycetka_gen_pdf.btn_odeslat_click%28object/) sender, EventArgs e)

I run it on Windows 11. Even when I install it on my developing machine with Windows 10 it throws same error. It only works when I run it throw Visual Studio... 

What I am missing that it fails :(

**Comment by MarcinZiabek** [comment]
Please provide an exact list of installed nuget packages with their versions. Also, please try to clear the environment and rebuild the solution.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
